src/OFS/tests/testRanges.py

```diff
@@ -186,15 +186,20 @@ def expectMultipleRanges(self, range, sets, draft=0):
 
         ct = rsp.getHeader('content-type').split(';')[0]
         draftprefix = draft and 'x-' or ''
-        self.assertFalse(ct != 'multipart/%sbyteranges' % draftprefix)
+        self.assertEqual(ct, 'multipart/%sbyteranges' % draftprefix)
         if rsp.getHeader('content-length'):
-            self.assertFalse(rsp.getHeader('content-length') != str(len(body)))
+            self.assertEqual(rsp.getHeader('content-length'), str(len(body)))
 
-        # Decode the multipart message
-        bodyfile = io.BytesIO('Content-Type: %s\n\n%s' % (
-            rsp.getHeader('content-type'), body))
-        partmessages = [part
-                        for part in email.message_from_file(bodyfile).walk()]
+        # Decode the multipart message and force a latin-1 encoding,
+        # revert that later after the email was parsed
+        bodyfile = io.StringIO(
```

Comment:
I thought so too. But when I tried the ` email.message_from_file` call just below with binary it blew up. Apparently that wants text. That's why I ended up with the forced latin-1 decode, get_payload encode trick.